### ISSUE TITLE
Requirements Issues

### ISSUE THREAD

**Issue by deusebio** [CGnal/python-ci-template#25]
Handling requirements is surely a nightmare. But we should find a guideline on how the requirements compilation should be handled and executed. 

At the present point we face an issue with the two types of requirements we have, prod and dev. Dev is generally the set of requirements which need to be installed in order to test, lint and check the code. One issue that came up is when the requirements of the package (prod) are not consistent with the ones used for testing. In this case, we cannot quite produce an environment where we perform tests & checks. Suppose prod depends on package A>3.6, the requirements in dev requires A<3.8 and last update of the package is 3.9. Compiling prod would produce (given A>3.6) an env with A==3.9. However, the conflict would only appear when compiling dev requirements. In this example, it is however easy to see that the most sensible choice would be to pick version 3.7, which satisfies all constraints. 

In short, it would be very cool if we were able to find a solution where the prod requirements are compiled by taking into account also the requirements of dev, but then only the requirements needed by prod would be exported.  

Or do you see other ways which promote stable and consistent environments?

**Comment by deusebio** [comment]
@nromeoarena @NicolaDonelli

**Comment by NicolaDonelli** [comment]
Why crack our heads on inventing something like that when the easiest solution would be to simply constrain dev requirements to prod requirements (solving conflicts when they rise, eventually not using dev tools that irreconcilably conflict with prod requirements)?
